src/types/Elements.ts

```diff
@@ -0,0 +1,12 @@
+import { Route } from 'nextjs-routes';
+import { AnchorHTMLAttributes } from 'react';
+
+export type AnchorElementProps = AnchorHTMLAttributes<HTMLAnchorElement>;
+
+export type ExternalAnchorElementProps = AnchorElementProps & {
+  href: string;
+};
+
+export type InternalAnchorElementProps = Omit<AnchorElementProps, 'href'> & {
+  href: Route;
+};
```

Comment:
@tbezman I wasn't sure of a good name for this, so went with something generic (`Elements.ts`) that could house similar types in the future.